### PR TITLE
Improve handling of alpha when saving to jpeg.

### PR DESCRIPTION
**Pull request by anntzer**
We can compose the figure facecolor against a white background before
rendering the image, rather than after.  This saves an image composition
step.

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

File: lib/matplotlib/backends/backend_agg.py [hunk 1]
```diff
@@ -563,23 +563,26 @@ def print_jpg(self, filename_or_obj, *args, dryrun=False, pil_kwargs=None,
             `PIL.Image.save` when saving the figure.  These take precedence
             over *quality*, *optimize* and *progressive*.
         """
-        FigureCanvasAgg.draw(self)
-        if dryrun:
-            return
         # The image is "pasted" onto a white background image to safely
```
Comment (timhoffm) [inline lib/matplotlib/backends/backend_agg.py]: The comment needs to be adapted.

Comment (anntzer) [reply lib/matplotlib/backends/backend_agg.py]: well, this is still pasting the figure over a white background.

File: lib/matplotlib/backends/backend_agg.py [hunk 2]
```diff
@@ -563,23 +563,25 @@ def print_jpg(self, filename_or_obj, *args, dryrun=False, pil_kwargs=None,
             `PIL.Image.save` when saving the figure.  These take precedence
             over *quality*, *optimize* and *progressive*.
         """
-        FigureCanvasAgg.draw(self)
+        # The image is "pasted" onto a white background to handle transparency.
```
Comment (timhoffm) [inline lib/matplotlib/backends/backend_agg.py]: I think the term "paste" is a bit misleading

```suggestion
        # Remove transparency by alpha-blending on an assumed white background
```

Comment (anntzer) [reply lib/matplotlib/backends/backend_agg.py]: sure, sounds good.

**Comment by QuLogic** [comment]
Will this cause weird flickering on interactive views (maybe not regular figures that composite on white, but embedded/notebook/something else)?

**Comment by anntzer** [comment]
I'm not sure why it would?

**Comment by QuLogic** [comment]
I was thinking of stuff like changing the DPI, but I see setting the canvas manager is handled in the call stack above this one.

**Pull request merged**